modules/component/src/core/cd-aware/strategy.ts

```diff
@@ -0,0 +1,339 @@
+// @Notice this part of the code is in the coalescing PR https://github.com/ngrx/platform/pull/2456
+// import { generateFrames } from '../projections/generateFrames';
+// import { coalesce } from '../operators/coalesce';
+import { defer, from, MonoTypeOperatorFunction, Observable } from 'rxjs';
+import {
+  ChangeDetectorRef,
+  NgZone,
+  ɵdetectChanges as detectChanges,
+  ɵmarkDirty as markDirty,
+} from '@angular/core';
+import { hasZone, isIvy } from '../utils';
+import { mapTo } from 'rxjs/operators';
+
+/** A shared promise instance to cause a delay of one microtask */
+let resolvedPromise: Promise<void> | null = null;
+
+function getResolvedPromise() {
+  resolvedPromise = resolvedPromise || Promise.resolve();
+  return resolvedPromise;
+}
+
+function getSaveDurationSelector(ngZone: NgZone): () => Observable<number> {
+  return () => defer(getResolvedPromise).pipe(mapTo(1));
+  // @Notice this part of the code is in the coalescing PR https://github.com/ngrx/platform/pull/2456
+  /* hasZone(ngZone)
+      ? generateFrames(
+          (window as any).__zone_symbol__requestAnimationFrame,
+          (window as any).__zone_symbol__cancelAnimationFrame
+        )
+      : generateFrames();*/
+}
+
+export interface StrategyFactoryConfig {
+  component: any;
+  ngZone: NgZone;
+  cdRef: ChangeDetectorRef;
+}
+
+export interface CdStrategy<T> {
+  behaviour: (cfg?: any) => MonoTypeOperatorFunction<T>;
+  render: () => void;
+  name: string;
+}
+
+export const DEFAULT_STRATEGY_NAME = 'idle';
+
+export interface StrategySelection<U> {
+  idle: CdStrategy<U>;
+
+  [key: string]: CdStrategy<U>;
+}
+
+export function getStrategies<T>(
+  cfg: StrategyFactoryConfig
+): StrategySelection<T> {
+  return {
+    idle: createIdleStrategy<T>(cfg),
+    noop: createNoopStrategy<T>(),
+    pessimistic1: createPessimistic1Strategy<T>(cfg),
+    pessimistic2: createPessimistic2Strategy<T>(cfg),
+    optimistic1: createOptimistic1Strategy<T>(cfg),
+    optimistic2: createOptimistic2Strategy<T>(cfg),
+  };
+}
+
+/**
+ * Idle Strategy
+ *
+ * This strategy is the drop-in replacement for Angular's built-in `async` pipe.
+ * This is the only strategy that **does not also work in zone-less environments**.
+ *
+ * - \>=ViewEngine
+ * |          | Render Method      | Coalescing | Coalesce Scope |
```

Comment:
@SanderElias suggests to change the naming form `Coalescing` and `Coalescing Scope`  to something developers can pick up fast. Coalescing is a word we all have to google so it will be hard to grasp.

Maybe `squash` or `squashing` like in `git squash` ?

Reply:
As the strategy apply the coalescing when needed we can skip it completely